src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -0,0 +1,1280 @@
+
+        self.use_shortcut = not config.true_skip
+        if self.use_shortcut:
+            self.shortcut = SConv1d(
+                dim,
+                dim,
+                kernel_size=1,
+                norm=config.norm,
+                norm_kwargs=config.norm_params,
+                causal=config.causal,
+                pad_mode=config.pad_mode,
+            )
+
+    def forward(self, x):
+        y = x
+        for layer in self.block:
+            y = layer(y)
+
+        if self.use_shortcut:
+            return self.shortcut(x) + y
+        else:
+            return y
+
+
+class EncodecEncoder(nn.Module):
+    """SEANet encoder."""
+
+    def __init__(self, config: EncodecConfig):
+        super().__init__()
+
+        act = getattr(nn, config.activation)
+        mult = 1
+        model = [
+            SConv1d(
+                config.audio_channels,
+                mult * config.num_filters,
+                config.kernel_size,
+                norm=config.norm,
+                norm_kwargs=config.norm_params,
+                causal=config.causal,
+                pad_mode=config.pad_mode,
+            )
+        ]
+
+        # Downsample to raw audio scale
+        for i, ratio in enumerate(reversed(config.ratios)):
+            # Add residual layers
+            for j in range(config.num_residual_layers):
+                model += [
+                    EncodecResnetBlock(
+                        config,
+                        mult * config.num_filters,
+                        kernel_sizes=[config.residual_kernel_size, 1],
+                        dilations=[config.dilation_base**j, 1],
+                    )
+                ]
+            # Add downsampling layers
+            model += [
+                act(**config.activation_params),
+                SConv1d(
+                    mult * config.num_filters,
+                    mult * config.num_filters * 2,
+                    kernel_size=ratio * 2,
+                    stride=ratio,
+                    norm=config.norm,
+                    norm_kwargs=config.norm_params,
+                    causal=config.causal,
+                    pad_mode=config.pad_mode,
+                ),
+            ]
+            mult *= 2
+
+        if config.lstm:
+            model += [SLSTM(mult * config.num_filters, num_layers=config.lstm)]
+
+        model += [
+            act(**config.activation_params),
+            SConv1d(
+                mult * config.num_filters,
+                config.dimension,
+                config.last_kernel_size,
+                norm=config.norm,
+                norm_kwargs=config.norm_params,
+                causal=config.causal,
+                pad_mode=config.pad_mode,
+            ),
+        ]
+
+        self.layers = nn.ModuleList(model)
+
+    def forward(self, x):
+        for layer in self.layers:
+            x = layer(x)
+        return x
+
+
+class EncodecDecoder(nn.Module):
+    """Encodec decoder."""
```

Comment:
I suggest we keep the comment here "SEANet decoder" since that's what it really is.